bitcoin/src/blockdata/transaction.rs

```diff
@@ -1142,9 +1132,27 @@ impl InputWeightPrediction {
 
     /// Computes the **signature weight** added to a transaction by an input with this weight prediction,
     /// not counting the prevout (txid, index), sequence, potential witness flag bytes or the witness count varint.
+    #[deprecated(since = "TBD", note = "use `InputWeightPrediction::witness_weight()` instead")]
     pub const fn weight(&self) -> Weight {
         Weight::from_wu_usize(self.script_size * 4 + self.witness_size)
     }
```

Comment:
You forgot to call through to the function mentioned in the deprecation message.

Reply:
I guess in this case that makes sense.  Thanks for the reminder.